File: Arrays and Hashing/705-Design-HashSet/705.py
class MyHashSet:

    def __init__(self):
        self.set_array = [[]] * 10000

    def add(self, key: int) -> None:
        idx = key % 10000
        if not key in self.set_array[idx]:
            self.set_array[idx].append(key)

    def remove(self, key: int) -> None:
        idx = key % 10000
        if key in self.set_array[idx]:
            self.set_array[idx].remove(key)

    def contains(self, key: int) -> bool:
        idx = key % 10000
        if key in self.set_array[idx]:
            return True
        else:
            return False


# Your MyHashSet object will be instantiated and called as such:
# obj = MyHashSet()
# obj.add(key)
# obj.remove(key)
# param_3 = obj.contains(key)

File: Arrays and Hashing/705-Design-HashSet/705.java
class MyHashSet {

    List<Integer>[] set_array;

    public MyHashSet() {
        set_array = new ArrayList[10000];
        for(int i = 0; i < 10000; i++){
            set_array[i] = new ArrayList<>();
        }
    }
    
    public void add(int key) {
        int idx = key % 10000;
        if(!set_array[idx].contains(key)){
            set_array[idx].add(key);
        }
    }
    
    public void remove(int key) {
        int idx = key % 10000;
        set_array[idx].remove(Integer.valueOf(key));
    }
    
    public boolean contains(int key) {
        int idx = key % 10000;
        return set_array[idx].contains(key);
    }
}

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet obj = new MyHashSet();
 * obj.add(key);
 * obj.remove(key);
 * boolean param_3 = obj.contains(key);
 */